homeassistant/components/syncthing/sensor.py

```diff
@@ -0,0 +1,224 @@
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    DOMAIN,
+    FOLDER_PAUSED_RECEIVED,
+    FOLDER_SENSOR_ALERT_ICON,
+    FOLDER_SENSOR_DEFAULT_ICON,
+    FOLDER_SENSOR_ICONS,
+    FOLDER_SUMMARY_RECEIVED,
+    SCAN_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    STATE_CHANGED_RECEIVED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Syncthing sensors."""
+
+    name = config_entry.data[CONF_NAME]
+    syncthing = hass.data[DOMAIN][name]
+
+    try:
+        config = await syncthing.system.config()
+        version = await syncthing.system.version()
+        dev = [
+            FolderSensor(
+                hass,
+                syncthing,
+                name,
+                folder["id"],
+                folder["label"],
+                version["version"],
+            )
+            for folder in config["folders"]
+        ]
+
+        async_add_entities(dev)
+    except aiosyncthing.exceptions.SyncthingError as exception:
+        raise PlatformNotReady from exception
+
+
+class FolderSensor(Entity):
+    """A Syncthing folder sensor."""
+
+    def __init__(self, hass, syncthing, name, folder_id, folder_label, version):
+        """Initialize the sensor."""
+        self.hass = hass
+        self._syncthing = syncthing
+        self._name = name
+        self._folder_id = folder_id
+        self._folder_label = folder_label
+        self._state = None
+        self._unsub_timer = None
+        self._version = version
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} {self._folder_id} {self._folder_label}"
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the entity."""
+        return f"{DOMAIN}-{self._name}-{self._folder_id}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state["state"]
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._state is not None
+
+    @property
+    def icon(self):
+        """Return the icon for this sensor."""
+        if self._state is None:
+            return FOLDER_SENSOR_DEFAULT_ICON
+        if self.state in FOLDER_SENSOR_ICONS:
+            return FOLDER_SENSOR_ICONS[self.state]
+        return FOLDER_SENSOR_ALERT_ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._state
+
+    @property
+    def should_poll(self):
+        """Return the polling requirement for this sensor."""
+        return False
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return {
+            "identifiers": {(DOMAIN, self._name)},
```

Comment:
We need to use something unique and stable as I'd, ie not name. The config entry entry_id could be an option.